cmd/create/idp/htpasswd.go

```diff
@@ -44,11 +44,13 @@ func createHTPasswdIDP(cmd *cobra.Command,
 	}
 
 	// Choose which way to create the IDP according to whether it already has an admin or not.
-	htpasswdIDP := FindExistingHTPasswdIDP(cluster, ocmClient)
+	htpasswdIDP, userList := FindExistingHTPasswdIDP(cluster, ocmClient)
 	if htpasswdIDP != nil {
-		containsAdminOnly := !HasClusterAdmin(htpasswdIDP) || htpasswdIDP.Htpasswd().Users().Len() == 0
+		containsAdminOnly := HasClusterAdmin(userList) && userList.Len() == 1
 		if !containsAdminOnly {
```

Comment:
is it  `if containsAdminOnly {`

Reply:
I added documentation for this line. 
The idea behind this is to allow users to call `rosa create admin` and afterwards call `rosa create idp` with type htpasswd without having the idp creation fail due to the preexisting idp they may have not known was created during `create admin`. If `!containsAdminOnly` we can infer that the idp was created via `ocm create idp` or `rosa create idp`, so logically creating a new htpasswd idp should fail. 
I hope it makes sense. We do intend to deprecate this command in the future since now htpasswd is a proper idp, this is only to maintain compatibility.